src/activity_actor.cpp

```diff
@@ -1184,10 +1184,18 @@ void consume_activity_actor::finish( player_activity &act, Character & )
     }
     //setting act to null clears these so back them up
     std::vector<int> temp_selections = consume_menu_selections;
-    act.set_to_null();
+    if( act.id() == activity_id( "ACT_CONSUME" ) ) {
```

Comment:
the comment is true, but i didn't want to mention spellcasting specifically because it would also override any other ongoing action that resulted from activating the item. i didn't know what other actions there were.